src/shims/posix/fs.rs

```diff
@@ -537,8 +602,9 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
     ) -> InterpResult<'tcx, i64> {
         let this = self.eval_context_mut();
 
-        this.check_no_isolation("write")?;
-        assert!(fd >= 3);
+        if fd >= 3 {
+            this.check_no_isolation("write")?;
```

Comment:
Move this to the `FileDescriptor<'tcx> for FileHandle` impl. The same with the `read` isolation check. I think right now you forbid reading from stdin with isolation

Reply:
The FileDescriptor methods won't have `this` so I don't think it'll be possible to do this check there.

For reads on `stdin`, @RalfJung suggested forbidding it with isolation: https://github.com/rust-lang/miri/pull/1436#pullrequestreview-419830068

Reply:
hmm.. .ok, let's keep forbidding it then

Reply:
Maybe they should have a `this` then? Or they should have a `const bool: ISOLATED_WRITES` or so.

The current code I think lets used escape isolation by using `dup` to put a normal file on file descriptor 0. We should add a test for that.

Reply:
Ah no, the lower FDs cannot be closed so isolation is still fine.